admin-core/src/main/kotlin/com/epam/drill/admin/jwt/config/JwtConfig.kt

```diff
@@ -21,28 +21,60 @@ import com.epam.drill.admin.config.*
 import com.epam.drill.admin.jwt.user.*
 import io.ktor.application.*
 import io.ktor.config.*
+import mu.KotlinLogging
 import java.util.*
+import javax.crypto.KeyGenerator
 import kotlin.time.*
 
-val Application.jwtConfig: ApplicationConfig
-    get() = environment.config.config("jwt")
+val logger = KotlinLogging.logger {}
+val Application.jwtProperties: ApplicationConfig
+    get() = drillConfig.config("jwt")
 
+val generatedSecret: String by lazy {
+    logger.warn { "The generated secret key for the JWT is used. " +
+            "To set your secret key, use the DRILL_JWT_SECRET environment variable." }
+    generateSecret()
+}
+val Application.jwtSecret: String
```

Comment:
Won't it be better to extend `JwtConfig` with those methods and not `Application`?

Reply:
`val Application.jwtProperties: ApplicationConfig get() = drillConfig.config("jwt")` dependes on Application instance - it'll be hard to change that